apps/coordinator/src/components/BCUR2/BCUR2Reader.tsx

```diff
@@ -1,136 +1,37 @@
-    if (!text || !text.toLowerCase().startsWith("ur:")) return;
-
-    try {
-      decoder.receivePart(text);
-
-      if (decoder.isComplete()) {
-        const extendedPublicKeyData = decoder.getDecodedData(network);
-        if (!extendedPublicKeyData)
-          throw new Error("Failed to decode extended public key data.");
-        if (!extendedPublicKeyData.bip32Path)
-          throw new Error(
-            "BIP32 path is missing in the extended public key data",
-          );
-
-        statusRef.current = "complete";
-        setIsScanning(false);
-
-        // Ensure the bip32Path starts with "m/"
-        const data = {
-          ...extendedPublicKeyData,
-          bip32Path: extendedPublicKeyData.bip32Path.startsWith("m/")
-            ? extendedPublicKeyData.bip32Path
-            : `m/${extendedPublicKeyData.bip32Path}`,
-        };
-
-        onSuccess(data);
-
-        decoder.reset();
-      }
-    } catch (e) {
-      console.error(e);
-      statusRef.current = "error";
-      setError(e instanceof Error ? e.message : String(e));
-      setIsScanning(false);
-      decoder.reset();
-    }
-  };
+interface BCUR2ReaderXPubProps extends BCUR2ReaderBaseProps {
+  mode: "xpub";
+  network: BitcoinNetwork;
+  onSuccess: (data: ExtendedPublicKeyData) => void;
+}
 
-  return (
-    <Box display="flex" flexDirection="column" alignItems="center">
-      {isScanning ? (
-        <>
-          <Paper elevation={3} sx={{ width, aspectRatio: "1" }}>
-            <QrReader
-              onResult={handleQRResult}
-              constraints={{ facingMode: "environment" }}
-              containerStyle={{ width: "100%", height: "100%" }}
-              scanDelay={200}
-            />
-          </Paper>
-          <Button
-            variant="outlined"
-            color="secondary"
-            onClick={handleStop}
-            sx={{ mt: 2 }}
-          >
-            Cancel
-          </Button>
-        </>
-      ) : (
-        <Button
-          variant="contained"
-          color="primary"
-          onClick={handleStart}
-          sx={{ mt: 2 }}
-        >
-          {startText}
-        </Button>
-      )}
+interface BCUR2ReaderPSBTProps extends BCUR2ReaderBaseProps {
+  mode: "psbt";
+  network?: BitcoinNetwork;
+  onSuccess: (psbt: string) => void;
+}
 
-      {error && (
-        <FormHelperText error sx={{ mt: 1 }}>
-          {error}
-        </FormHelperText>
-      )}
+type BCUR2ReaderProps = BCUR2ReaderXPubProps | BCUR2ReaderPSBTProps;
 
-      {!error && isScanning && (
-        <Typography variant="body2" sx={{ mt: 1 }}>
-          Scanning... Show all QR parts in sequence.
-        </Typography>
-      )}
-    </Box>
-  );
+/**
+ * Wrapper component that renders the appropriate BCUR2 reader based on mode.
+ * TypeScript ensures type safety for the onSuccess callback based on the mode.
+ */
+const BCUR2Reader: React.FC<BCUR2ReaderProps> = (props) => {
+  return <BCUR2Scanner {...props} />;
```

Comment:
I'm not clear now what the difference between BCUR2Reader and BCUR2Scanner is. they both have the same prop types, and the reader seems to just return the scanner. Seems like this file isn't needed anymore.

Reply:
Or the scanner code could just be moved here.